paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/HiveSchema.java

```diff
@@ -233,9 +233,10 @@ private static void checkFieldsMatched(
             }
         }
 
-        if (schemaFieldNames.size() != hiveFieldNames.size()) {
+        // It is OK that hive is a subset of paimon
+        if (schemaFieldNames.size() < hiveFieldNames.size()) {
```

Comment:
Checking schema filed names is a subset of paimon's just by List#size() is not precise. Use Set#containsAll() maybe better. :)

Reply:
The following code will be checked one by one, I think it is OK